test/spec/auctionmanager_spec.js

```diff
@@ -852,66 +852,54 @@ describe('auctionmanager.js', function () {
         config.resetConfig();
       });
 
-      it('are dropped after their last bid becomes stale (if minBidCacheTTL is set)', () => {
+      it('are dropped after `minBidCacheTTL` seconds if they had no bid', () => {
+        auction.callBids();
         config.setConfig({
-          minBidCacheTTL: 0
+          minBidCacheTTL: 2
         });
+        return auction.end.then(() => {
+          expect(auctionManager.getNoBids().length).to.eql(1);
+          clock.tick(10 * 10000);
+          expect(auctionManager.getNoBids().length).to.eql(0);
+        })
+      });
+    });
+
+    describe('stale bids', () => {
+      let clock, auction;
+      beforeEach(() => {
+        clock = sinon.useFakeTimers();
+        auction = auctionManager.createAuction({adUnits});
+        indexAuctions.push(auction);
+      });
+      afterEach(() => {
+        clock.restore();
+        config.resetConfig();
+      });
+
+      it('are dropped after they expire', () => {
         bids = [
           {
+            adId: '1',
             adUnitCode: ADUNIT_CODE,
             adUnitId: ADUNIT_CODE,
             ttl: 10
           }, {
+            adId: '2',
             adUnitCode: ADUNIT_CODE,
             adUnitId: ADUNIT_CODE,
             ttl: 100
           }
         ];
         auction.callBids();
         return auction.end.then(() => {
-          clock.tick(50 * 1000);
-          expect(auctionManager.getBidsReceived().length).to.equal(2);
-          clock.tick(56 * 1000);
+          clock.tick(10 * 1000);
+          expect(auctionManager.getBidsReceived().length).to.equal(1);
+          clock.tick(90 * 1000);
           expect(auctionManager.getBidsReceived()).to.eql([]);
         });
       });
-
-      it('are dropped after `minBidCacheTTL` seconds if they had no bid', () => {
-        auction.callBids();
-        config.setConfig({
-          minBidCacheTTL: 2
-        });
-        return auction.end.then(() => {
-          expect(auctionManager.getNoBids().length).to.eql(1);
-          clock.tick(10 * 10000);
-          expect(auctionManager.getNoBids().length).to.eql(0);
-        })
-      });
-
-      Object.entries({
-        'bids': {
-          bd: [{
-            adUnitCode: ADUNIT_CODE,
-            adUnitId: ADUNIT_CODE,
-            ttl: 10
-          }],
-          entries: () => auctionManager.getBidsReceived()
-        },
-        'no bids': {
-          bd: [],
-          entries: () => auctionManager.getNoBids()
-        }
-      }).forEach(([t, {bd, entries}]) => {
-        it(`with ${t} are never dropped if minBidCacheTTL is not set`, () => {
```

Comment:
would it be better to keep this behavior? expire individual bids the same way auctions are, after `Math.max(bid.ttl, getConfig('minBidCacheTTL'))`, and only if `minBidCacheTTL` is set. This way pubs can choose whether they want this / they don't have to reason about two different expiration mechanisms.

Reply:
@fowler446 are you still working on this? Can you provide your perspective on this?